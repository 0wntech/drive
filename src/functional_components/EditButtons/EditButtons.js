import React from 'react';
import PropTypes from 'prop-types';
import ActionButton from '../ActionButton';
import useWindowDimension from '../../hooks/useWindowDimension';
import Check from '../../assets/svgIcons/Check';
import X from '../../assets/svgIcons/X';
import EditIcon from '../../assets/svgIcons/Edit';
import styles from './EditButtons.module.scss';
import styleConstants from '../../styles/constants.scss';

const EditButtons = ({ isEditable, onSubmit, onCancel, onEdit }) => {
    const { width } = useWindowDimension();
    if (isEditable) {
        if (width < styleConstants.screen_m) {
            return (
                <div className={styles.editButtons}>
                    <ActionButton
                        className={styles.actionButton}
                        onClick={onCancel}
                        label="Cancel"
                        size="lg"
                        color="red"
                        dataId="edit-cancel"
                    />
                    <ActionButton
                        className={styles.actionButton}
                        onClick={onSubmit}
                        label="Save"
                        size="lg"
                        color="green"
                        dataIid="edit-submit"
                    />
                </div>
            );
        } else if (width > styleConstants.screen_m) {
            return (
                <div className={styles.editButtons}>
                    <X
                        onClick={onCancel}
                        className={styles.icon}
                        data-test-id="edit-cancel"
                    />{' '}
                    <Check
                        className={styles.icon}
                        onClick={onSubmit}
                        data-test-id="edit-submit"
                    />
                </div>
            );
        }
    } else {
        if (width < styleConstants.screen_m) {
            return (
                <ActionButton onClick={onEdit} size="lg" color="blue">
                    <EditIcon
                        viewBox="3 2 30 30"
                        width="20"
                        height="20"
                        className={styles.iconWhite}
                        data-test-id="edit"
                    />
                    Edit Profile
                </ActionButton>
            );
        } else {
            return (
                <EditIcon
                    onClick={onEdit}
                    className={styles.icon}
                    data-test-id="edit"
                />
            );
        }
    }
};

EditButtons.propTypes = {
    isEditable: PropTypes.bool,
    onSave: PropTypes.func,
    onCancel: PropTypes.func,
    onEdit: PropTypes.func,
};

export default EditButtons;
