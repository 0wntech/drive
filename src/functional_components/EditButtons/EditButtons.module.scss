.editButtons {
    display: flex;
    flex-direction: column;
}

.actionButton {
    margin: 0 4px;
}

.icon {
    cursor: pointer;
}
