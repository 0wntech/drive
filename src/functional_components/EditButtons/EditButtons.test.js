import React from 'react';
import EditButtons from './EditButtons';
import TestRenderer from 'react-test-renderer';

describe('EditButtons', () => {
    describe('EditButtons Snapshot', () => {
        it('matches the snapshot', () => {
            const tree = TestRenderer.create(<EditButtons />);
            expect(tree.toJSON()).toMatchSnapshot();
        });
    });
});
