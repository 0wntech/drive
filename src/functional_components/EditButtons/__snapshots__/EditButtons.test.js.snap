// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditButtons EditButtons Snapshot matches the snapshot 1`] = `
<svg
  className="icon"
  data-test-id="edit"
  fill="none"
  height={32}
  width={32}
>
  <path
    clipRule="evenodd"
    d="M22.267 3.067l6.666 6.666a1.289 1.289 0 010 1.867L11.6 28.933c-.267.267-.533.4-.933.4H4c-.8 0-1.333-.533-1.333-1.333v-6.667c0-.4.133-.666.4-.933L20.4 3.067a1.289 1.289 0 011.867 0zm-16.934 23.6h4.8l16-16-4.8-4.8-16 16v4.8z"
    fill="#000"
    fillRule="evenodd"
  />
  <mask
    height={28}
    id="edit_svg__a"
    maskUnits="userSpaceOnUse"
    width={28}
    x={2}
    y={2}
  >
    <path
      clipRule="evenodd"
      d="M22.267 3.067l6.666 6.666a1.289 1.289 0 010 1.867L11.6 28.933c-.267.267-.533.4-.933.4H4c-.8 0-1.333-.533-1.333-1.333v-6.667c0-.4.133-.666.4-.933L20.4 3.067a1.289 1.289 0 011.867 0zm-16.934 23.6h4.8l16-16-4.8-4.8-16 16v4.8z"
      fill="#fff"
      fillRule="evenodd"
    />
  </mask>
  <g
    mask="url(#edit_svg__a)"
  >
    <path
      clipRule="evenodd"
      d="M0 0h32v32H0V0z"
      fill="#000"
      fillRule="evenodd"
    />
  </g>
</svg>
`;
